src/main.rs

```diff
@@ -585,80 +586,100 @@ impl ManParser for Type1 {
-//         if options_matched == None:
-//             add_diagnostic('Still not found')
-//             return False
-//         while options_matched != None:
-//             data = options_matched.group(2)
-//             data = remove_groff_formatting(data)
-//             data = data.strip()
-//             data = data.split("\n",1)
-//             if (len(data)>1 and len(data[1].strip())>0): # and len(data[1])<400):
-//                 optionName = data[0].strip()
-//                 if ( optionName.find("-") == -1):
-//                     add_diagnostic("%r doesn't contain '-'" % optionName)
-//                 else:
-//                     optionName = unquote_double_quotes(optionName)
-//                     optionName = unquote_single_quotes(optionName)
-//                     optionDescription = data[1].strip().replace("\n"," ")
-//                     built_command(optionName, optionDescription)
-//             else:
-//                 add_diagnostic('Unable to split option from description')
-//                 return False
-//
-//             options_section = options_section[options_matched.end()-3:]
-//             options_matched = re.search(options_parts_regex, options_section)
-//         return True
-
 impl Type1 {
-    fn fallback(&self, options_section: &str) -> bool {
-        unimplemented!()
+    fn fallback(
+        &self,
+        mut options_section: &str,
+        built_command_output: &mut Vec<String>,
+        existing_options: &mut HashSet<String>,
+        cmd_name: &str,
+    ) -> bool {
+        // add_diagnostic("Trying fallback");
+        let options_parts_re = Regex::new(r"\.TP( \d+)?((?s:.)*?)\.TP").unwrap();
+        let mut options_matched = options_parts_re.captures(options_section);
+        if options_matched.is_none() {
+            // add_diagnostic("Still not found");
+            return false;
+        }
+        while let Some(mat) = options_matched {
+            let data = mat.get(2).unwrap().as_str();
+            let data = remove_groff_formatting(data);
+            let data: Vec<&str> = data.trim().splitn(2, '\n').collect();
+            if data.len() < 2 || data[1].trim().is_empty() {
+                // add_diagnostic("Unable to split option from description");
+                return false;
+            }
+            let option_name = data[0].trim();
+            if option_name.find('-').is_some() {
+                let option_name = unquote_double_quotes(option_name);
+                let option_name = unquote_single_quotes(option_name);
+                let option_desc = data[1].trim().replace('\n', " ");
+                built_command(
+                    option_name,
+                    option_desc.as_str(),
+                    built_command_output,
+                    existing_options,
+                    cmd_name,
+                );
+            } else {
+                // add_diagnostic(format!("{:?} does not contain '-'", option_name));
+                return false;
```

Comment:
Hmm. Why doesn't the original enter an infinite loop in this case, since it doesn't return on this branch? Is there mutation in the Python source that I'm not recognizing?

Reply:
Oh, yes. I probably got confused with https://github.com/rust-dc/fish-manpage-completions/pull/72/files#diff-639fbc4ef05b315af92b4d836c31b023L644 and https://github.com/rust-dc/fish-manpage-completions/pull/72/files#diff-639fbc4ef05b315af92b4d836c31b023L653.

Yeah, it should not return. In fact, this block is not required without the `add_diagnostic`.